### DISCUSSION TITLE
How to update upsc without losing date/settings?

### DISCUSSION THREAD

**Discussion by bheeshmpita** [tomasklaen/uosc#848] [Q&A] [answered]
Hi, i have uosc v4.2 and upgraded to latest one from the zip file but on doing so i dont see any ui which may be due to some settings. How can i update without losing my existing settings?

**Comment by christoph-heinrich** [comment, marked as answer]
The fact that you don't see anything makes me wonder if the script gets loaded properly.

Do you have a `~~/scripts/uosc/` directory, which then contains a `main.lua` and some directories?

And if so, have you looked at the console output? Maybe it crashes while loading for some reason.

Also the uosc.conf has changed a lot since v4.2, so you might want to consider starting over with a new one from the latest release (and then maybe copy parts of your old one over once things are working).

**Reply by bheeshmpita** [reply]
Thanks for the help. Looking from the previous directory structure which only contained one file named uosc.lua so i only moved main.lua this time. Following your instructions everything works fine and just conf file needs to be modified.
